server/middleware.js

```diff
@@ -82,15 +116,85 @@ module.exports.flux = {
    *
    * Use store actions and listeners to inflate the store.
    *
+   * The advantages of this approach are:
+   *
+   * - Uses the _exact same_ series of actions to inflate store as client.
+   *
+   * The disadvantages of this approach are:
+   *
+   * - Adds extra listeners in a slightly complicated way.
+   * - Cannot use flux singletons.
+   *
+   * **Flux Instance**: This middleware creates ephemeral flux instances to
+   * allow async actions free reign to mutate store state before snapshotting.
+   * The work sequence is:
+   *
+   * - Create new `flux` instance.
+   * - Set `ActionListeners` on appropriate events.
+   * - Invoke the necessary action(s) to inflate the store.
+   * - Snapshot the store data.
+   * - Clean up the flux instance, listeners, etc.
+   *
+   * @param   {Object}    Component React component to render.
    * @returns {Function}            middleware function
    */
-  actions: function (/*TODO: Component*/) {
-    return function (/*TODO: req, res, next*/) {
-      // TODO: NOTE -- Doesn't look safe! Setting listeners on the singleton
-      // store means potentially race conditions (?).
+  actions: function (Component) {
+    return function (req, res, next) {
+      // Check query string.
+      var queryBootstrap = _getQueryBootstrap(req);
+      if (!queryBootstrap) { return next(); }
+      var types = queryBootstrap.types;
+      var value = queryBootstrap.value;
+
+      // Flux instance for this single request / callback.
+      var flux = new Flux();
+      var listener = new ActionListeners(flux);
+      var actions = flux.getActions("ConvertActions");
+
+      // Wrap cleanup methods.
+      var _done = function (err) {
+        listener.removeAllActionListeners();
+        flux.flush();
```

Comment:
No, I guess not! I was more trying to make sure nothing sticks around explicitly. But a simple `delete` can probably take care of that...